### PR TITLE
Missions and Maps Updates

### PR DESCRIPTION
**Pull request by oceanzus**

### CODE REVIEW

**Comment by oceanzus** [comment]
Fixing the conflicts and squashing commit history.  Please hold.

**Comment by oceanzus** [comment]
Ready for review.

File: src/web/central_command/client/components/AXUI.jsx [hunk 1]
```diff
@@ -606,10 +626,10 @@ export default class AXUI extends React.Component {
 		this.surveyPolygonInteraction.on(
 			'drawstart',
 			(evt) => {
-				this.setState({ surveyPolygonFeature: evt.feature });
+				this.setState({ surveyPolygonFeature: evt.feature, surveyPolygonChanged: false });
 
 				surveyPolygonlistener = evt.feature.getGeometry().on('change', (evt2) => {
-					console.log(evt);
+					// console.log(evt);
```
Comment (ferrom-jaia) [inline src/web/central_command/client/components/AXUI.jsx:631]: Do we want this console.log still?

File: src/web/central_command/client/components/AXUI.jsx [hunk 2]
```diff
@@ -1323,11 +1344,18 @@ export default class AXUI extends React.Component {
...
+		let goalSettingsPanel = '';
 		if (this.state.goalBeingEdited != null) {
 			goalSettingsPanel = <GoalSettingsPanel goal={this.state.goalBeingEdited} onChange={() => {this.updateMissionLayer()}} onClose={() => { this.state.goalBeingEdited = null }} />
 		}
 
+		// Add mission generation form to UI if the survey polygon has changed.
+		let missionSettingsPanel = '';
+		if (this.state.surveyPolygonChanged) {
+			missionSettingsPanel = <MissionSettingsPanel mission_params={this.state.missionParams} onClose={() => { this.generateMissions(this.state.surveyPolygonGeoCoords); this.state.surveyPolygonChanged = false }} />
+			// missionSettingsPanel = <MissionSettingsPanel mission_params={this.state.missionParams} onChange={() => {this.generateMissions(this.state.surveyPolygonGeoCoords)}} onClose={() => { this.state.surveyPolygonChanged = false }} />
```
Comment (ferrom-jaia) [inline src/web/central_command/client/components/AXUI.jsx:1355]: Can we delete this commented out line?

File: src/web/central_command/client/components/AXUI.jsx [hunk 3]
```diff
@@ -1667,18 +1703,22 @@ export default class AXUI extends React.Component {
...
 			}
 
+
 			let goals = missions[botId]?.plan?.goal || []
 
 			let transformed_pts = goals.map((goal) => {
 				return equirectangular_to_mercator([goal.location.lon, goal.location.lat])
 			})
 
+			// console.log('transformed_pts');
```
Comment (ferrom-jaia) [inline src/web/central_command/client/components/AXUI.jsx:1712]: Are these for debugging? If so, maybe we just add a comment above to explain what they provide.

File: src/web/central_command/client/components/MissionSettings.jsx [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    componentDidUpdate() {
+        this.onChange?.()
+    }
+
+    render() {
+        self = this
+
+        // var taskOptionsPanel
```
Comment (ferrom-jaia) [inline src/web/central_command/client/components/MissionSettings.jsx:29]: Can we delete these commented out lines?

File: src/web/central_command/client/components/AXUI.jsx [hunk 5]
```diff
@@ -30,6 +31,9 @@ import OlIcon from 'ol/style/Icon'
 import OlLayerGroup from 'ol/layer/Group';
 import OlSourceOsm from 'ol/source/OSM';
 import OlSourceXYZ from 'ol/source/XYZ';
+import OlTileWMS from 'ol/source/TileWMS';
```
Comment (ferrom-jaia) [inline src/web/central_command/client/components/AXUI.jsx:34]: I got an error while trying to run the run.sh script because this line is duplicated (line 34, and line 42)

File: src/web/central_command/client/components/AXUI.jsx [hunk 6]
```diff
@@ -315,31 +324,42 @@ export default class AXUI extends React.Component {
...
+                        }),
+                        new OlTileLayer({
+                                title: 'OpenStreetMap',
+                                //type: 'base',
+                                zIndex: 9,
+                                opacity: 0.3,
+                                source: new OlSourceOsm()
+                        }),
+                        new OlTileLayer({
+                                title: 'NOAA ENC Charts',
```
Comment (ferrom-jaia) [inline src/web/central_command/client/components/AXUI.jsx:342]: I tried turning this layer on but the layer did not show

Comment (edsanville) [reply src/web/central_command/client/components/AXUI.jsx:342]: Strange...  it loads for me!

**Comment by edsanville** [comment]
After removing the duplicate import, this Pull Request works for me.

File: src/web/central_command/client/components/AXUI.jsx (hunk 6 above)
Comment (ferrom-jaia) [reply src/web/central_command/client/components/AXUI.jsx:342]: Hmm let me try again

**Review by edsanville — APPROVED** [review]

File: src/web/central_command/client/components/AXUI.jsx (hunk 6 above)
Comment (ferrom-jaia) [reply src/web/central_command/client/components/AXUI.jsx:342]: Works for me as well! Does the open street maps layer looks darker now for you Ed?
